### PR TITLE
Dns settings refactor

### PR DESCRIPTION
**Pull request by rdubrock**
I notice that the DNS validation fields were tilting a bit, and thought that might be a nice first PR. I also took the opportunity to refactor that component a bit and set up some tests using react testing library. This PR is very much just trying things out and getting familiar with the tooling, any and all feedback very welcome!.

Before:

Screen Shot 2020-07-29 at 8 05 18 PM

After:

Screen Shot 2020-07-29 at 5 13 59 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,10 +13,14 @@
   "devDependencies": {
     "@grafana/data": "7.0.6",
     "@grafana/runtime": "7.0.6",
-    "@grafana/toolkit": "7.0.6",
```
Comment (rdubrock) [inline package.json:16]: Not sure we need this, I just gave it a go to see if that version of toolkit included a jest version > 24 (doesn't seem to)

**Comment by woodsaj** [comment]
This looks good to me.

the next step to improving this form and other validation forms that are similar, is to reduce the repitition.
ie, instead of showing the user 6 lists of validation rules, we can make it a single list, and allow users to dynamically select which field is being validated, and whether the expression should be inverted.

eg
image

**Comment by woodsaj** [comment]
To be clear, i think this PR should be merged, and further improvements applied in future PRs.

File: package.json (hunk 1 above)
Comment (ryantxu) [reply package.json:16]: sounds like we should make sure it does in the future!

**Pull request merged**